en/docs/develop/spring-boot.md

```diff
@@ -0,0 +1,399 @@
+
+       }
+
+       @Autowired
+       private ClientRegistrationRepository clientRegistrationRepository;
+
+       private LogoutSuccessHandler oidcLogoutSuccessHandler() {
+           OidcClientInitiatedLogoutSuccessHandler oidcLogoutSuccessHandler =
+                   new OidcClientInitiatedLogoutSuccessHandler(
+                           this.clientRegistrationRepository);
+
+           oidcLogoutSuccessHandler.setPostLogoutRedirectUri(
+                   URI.create("http://localhost:8080/oauth-login")); //Need to give the post-rediret-uri here
+
+           return oidcLogoutSuccessHandler;
+       }
+    }
+    ```
+    
+2. Create a template at `src/main/resources/login.html` to display the logout option and have a logout button.
+
+
+3.  Add the `/logout` redirection when user clicks the Logout button. 
+
+    ```html
+    <div style="float:right">
+       <form method="post" th:action="@{/logout}"  class="navbar-form navbar-right">
+           <button id="logout-button" type="submit" class="btn btn-danger">Logout</button>
+       </form>
+    </div>
+
+    ```
+
+### Customize the logout endpoint
+
+If you wish to customize the logout endpoint, follow the steps below. 
+
+1. Have a ConfigSecurity class by extending WebSecurityConfigurerAdapter and configure the logoutUrl
+    ```java
+
+
+    @EnableWebSecurity
+    public class ConfigSecurity extends WebSecurityConfigurerAdapter {
+
+       protected void configure(HttpSecurity http) throws Exception {
+
+           http.authorizeRequests()
+                   .antMatchers("/login")
+                   .permitAll()
+                   .anyRequest()
+                   .authenticated()
+                   .and().oauth2Login().loginPage("/login")
+                   .and().logout().logoutUrl("/applogout)
+                   .logoutSuccessHandler(oidcLogoutSuccessHandler());
+
+       }
+    @Autowired
+       private ClientRegistrationRepository clientRegistrationRepository;
+
+       private LogoutSuccessHandler oidcLogoutSuccessHandler() {
+           OidcClientInitiatedLogoutSuccessHandler oidcLogoutSuccessHandler =
+                   new OidcClientInitiatedLogoutSuccessHandler(
+                           this.clientRegistrationRepository);
+
+           oidcLogoutSuccessHandler.setPostLogoutRedirectUri(
+                   URI.create("http://localhost:8080/sprinb-boot-app"));
+
+           return oidcLogoutSuccessHandler;
+       }
+
+    }
+
+    ```
+####  Have a Logout button and redirect to custom logout (“/applogout”) url
+ 
+Create a template at `src/main/resources/login.html` to display the Logout option and have a logout button.
+
+Add the `/applogout` redirection when user clicks the `Logout` button.  
+
+```html
+
+<div style="float:right">
+   <form method="post" th:action="@{/applogout}"  class="navbar-form navbar-right">
+       <button id="logout-button" type="submit" class="btn btn-danger">Logout</button>
+   </form>
+</div>
+
+```
```

Comment:
shall we add numbering for the steps?

Reply:
Have added steps if there are many. These have only one step and the sub sections are not in sequence so can't make them a list.